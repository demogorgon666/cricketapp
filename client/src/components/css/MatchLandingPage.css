@import url('https://fonts.googleapis.com/css?family=Montserrat&display=swap');

body{
    font-family: 'Montserrat', sans-serif;
    color: #272727;
}

.timeline{
    width: 90%;
    height: 80px;
    padding: 5px;
    border: 1px solid black;
    border-radius: 8px;
    margin: 15px;
}

.all-recent-matches-box{
    display: grid;
    grid-template-columns: 22vw 22vw 22vw 22vw;
    width: 90%;
    height: 310px;
    /* border: 1px solid black; */
    border-radius: 8px;
    margin: 10px;

}
.inside-recent-matches-box{
    display: flex;
    flex-direction: column;
    margin: 4px;
    margin-bottom: 10px;
    width: 95%;
    height: 140px;
    border-radius: 8px;
    border: 1px solid black;
}
.inside-recent-matches-box:hover{
    background-color: rgb(116, 116, 158);
    width: 98%;
    height: 145px;
}



.inside-recent-matches-box .Team-data {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    text-align: left;
    font-size: 10pt;
    margin: 3px;
    margin-left: 15px;
}

.inside-recent-matches-box .tournamnet-name {
    text-align: left;
    font-size: 15pt;
    margin-left: 10pt;
    margin-top: 10pt;
}


.inside-recent-matches-box .TeamOne-name{
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 12pt;
}
.inside-recent-matches-box  .TeamTwo-name {
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 12pt;
}

.inside-recent-matches-box  .TeamOne-score{
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 10pt;
    margin-right: 25pt;
}    
.inside-recent-matches-box  .TeamTwo-score{
    text-align: left;
    font-size: 13pt;
    font-weight: 500;
    margin-left: 10pt;
    margin-right: 25pt;
}    

.inside-recent-matches-box .winner-name{
    text-align: left;
    font-size: 16px;;
    font-weight: 600;
    margin-left: 10px;
    margin-top: 5pt;
}


@media only screen and (max-width: 1230px) {

    .all-recent-matches-box{
        display: grid;
        grid-template-columns: 23vw 23vw 23vw 23vw;
        width: 90%;
        height: 310px;
        /* border: 1px solid black; */
        border-radius: 8px;
        margin: 20px;
    
    }
    .inside-recent-matches-box{
        display: flex;
        flex-direction: column;
        margin: 4px;
        margin-bottom: 10px;
        width: 95%;
        height: 140px;
        border-radius: 8px;
        border: 1px solid black;
    }
    .inside-recent-matches-box:hover{
        background-color: rgb(116, 116, 158);
        width: 98%;
        height: 145px;
    }

    

    .inside-recent-matches-box .Team-data {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        text-align: left;
        font-size: 8pt;
        margin: 1px;
        margin-left: 15px;
    }

    .inside-recent-matches-box .tournamnet-name {
        text-align: left;
        font-size: 12pt;
        margin-left: 10pt;
        margin-top: 10pt;
    }
	

    .inside-recent-matches-box .TeamOne-name{
        text-align: left;
        font-size: 12pt;
        font-weight: 500;
        margin-left: 10pt;
    }
    .inside-recent-matches-box  .TeamTwo-name {
        text-align: left;
        font-size: 12pt;
        font-weight: 500;
        margin-left: 10pt;
    }

    .inside-recent-matches-box  .TeamOne-score{
        text-align: left;
        font-size: 12pt;
        font-weight: 500;
        margin-left: 8pt;
        margin-right: 15pt;
    }    
    .inside-recent-matches-box  .TeamTwo-score{
        text-align: left;
        font-size: 12pt;
        font-weight: 500;
        margin-left: 8pt;
        margin-right: 15pt;
    }    
   
    .inside-recent-matches-box .winner-name{
        text-align: left;
        font-size: 15px;;
        font-weight: 600;
        margin-left: 10pt;
        margin-top: 5pt;
    }

}




@media only screen and (max-width: 900px) {

    .all-recent-matches-box{
        display: grid;
        grid-template-columns: 45vw 45vw;
        width: 90%;
        height: 310px;
        /* border: 1px solid black; */
        border-radius: 8px;
        margin: 20px;
    
    }
    .inside-recent-matches-box{
        display: flex;
        flex-direction: column;
        margin: 4px;
        margin-bottom: 10px;
        width: 95%;
        height: 140px;
        border-radius: 8px;
        border: 1px solid black;
    }
    .inside-recent-matches-box:hover{
        background-color: rgb(116, 116, 158);
        width: 98%;
        height: 145px;
    }

    

    .inside-recent-matches-box .Team-data {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        text-align: left;
        font-size: 10pt;
        margin: 1px;
        margin-left: 15px;
    }

    .inside-recent-matches-box .tournamnet-name {
        text-align: left;
        font-size: 15pt;
        margin-left: 12pt;
        margin-top: 12pt;
    }
	

    .inside-recent-matches-box .TeamOne-name{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 15pt;
    }
    .inside-recent-matches-box  .TeamTwo-name {
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 15pt;
    }

    .inside-recent-matches-box  .TeamOne-score{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 30pt;
        margin-right: 50pt;
    }    
    .inside-recent-matches-box  .TeamTwo-score{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 30pt;
        margin-right: 50pt;
    }    
   
    .inside-recent-matches-box .winner-name{
        text-align: left;
        font-size: 16px;;
        font-weight: 600;
        margin-left: 10px;
        margin-top: 7pt;
    }

}


@media only screen and (max-width: 700px) {

    .all-recent-matches-box{
        display: grid;
        grid-template-columns: 45vw 45vw;
        width: 90%;
        height: 310px;
        /* border: 1px solid black; */
        border-radius: 8px;
        margin: 20px;
    
    }
    .inside-recent-matches-box{
        display: flex;
        flex-direction: column;
        margin: 4px;
        margin-bottom: 10px;
        width: 95%;
        height: 140px;
        border-radius: 8px;
        border: 1px solid black;
    }
    .inside-recent-matches-box:hover{
        background-color: rgb(116, 116, 158);
        width: 98%;
        height: 145px;
    }

    

    .inside-recent-matches-box .Team-data {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        text-align: left;
        font-size: 10pt;
        margin: 1px;
        margin-left: 15px;
    }

    .inside-recent-matches-box .tournamnet-name {
        text-align: left;
        font-size: 12pt;    
        margin-left: 10pt;
        margin-top: 10pt;
    }
	

    .inside-recent-matches-box .TeamOne-name{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 11pt;
    }
    .inside-recent-matches-box  .TeamTwo-name {
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 11pt;
    }

    .inside-recent-matches-box  .TeamOne-score{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 2pt;
        margin-right: 0pt;
    }    
    .inside-recent-matches-box  .TeamTwo-score{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 2pt;
        margin-right: 0pt;
    }    
   
    .inside-recent-matches-box .winner-name{
        text-align: left;
        font-size: 16px;;
        font-weight: 600;
        margin-left: 10px;
        margin-top: 5pt;
    }

}


@media only screen and (max-width: 500px) {

    .all-recent-matches-box{
        display: grid;
        grid-template-columns: 90vw;
        width: 90%;
        height: 310px;
        /* border: 1px solid black; */
        border-radius: 8px;
        margin: 20px;
    
    }
    .inside-recent-matches-box{
        display: flex;
        flex-direction: column;
        margin: 4px;
        margin-bottom: 10px;
        width: 95%;
        height: 140px;
        border-radius: 8px;
        border: 1px solid black;
    }
    .inside-recent-matches-box:hover{
        background-color: rgb(116, 116, 158);
        width: 98%;
        height: 145px;
    }

    

    .inside-recent-matches-box .Team-data {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        text-align: left;
        font-size: 10pt;
        margin: 1px;
        margin-left: 15px;
    }

    .inside-recent-matches-box .tournamnet-name {
        text-align: left;
        font-size: 12pt;    
        margin-left: 10pt;
        margin-top: 10pt;
    }
	

    .inside-recent-matches-box .TeamOne-name{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 11pt;
    }
    .inside-recent-matches-box  .TeamTwo-name {
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 11pt;
    }

    .inside-recent-matches-box  .TeamOne-score{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 2pt;
        margin-right: 20pt;
    }    
    .inside-recent-matches-box  .TeamTwo-score{
        text-align: left;
        font-size: 13pt;
        font-weight: 500;
        margin-left: 2pt;
        margin-right: 20pt;
    }    
   
    .inside-recent-matches-box .winner-name{
        text-align: left;
        font-size: 16px;;
        font-weight: 600;
        margin-left: 10px;
        margin-top: 15pt;
    }

}