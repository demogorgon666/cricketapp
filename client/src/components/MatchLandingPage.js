import React, { Component } from "react";
import { connect } from "react-redux";
import "./css/MatchLandingPage.css";
import Navbar from "../components/common/Navbar";
import { getRecentMatches, getMatchesDate } from "../actions/Matches.js";
import { Calendar } from '@y0c/react-datepicker';
import '@y0c/react-datepicker/assets/styles/calendar.scss';
import Loader from "react-loader-spinner";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

export class MatchLandingPage extends Component {

    // [Yatin] On selection of new date from the calender, 
    // new date will be passed to the match api to check 
    // if any matches were played on selected date

    onChange = (date) => {
        //[yatin] storing the selected date in date format
        var date = date.toDate();

        //[yatin] slicing the date in "yyyy-mm-dd" fromat
        //  before passing it to the match api function
        var datee = date.toJSON().slice(0, 10)

        //[yatin] passing the new selected date to the api
        this.props.getRecentMatches(datee);
    }
    componentDidMount() {
        this.props.getRecentMatches("2016-05-20");
    }

    render() {
        return (
            <div>
                <div>
                    <Navbar />
                </div>

                <div style={{ marginTop: "80px" }}>
                    <div>
                        <h1 className="h1-match" style={{ textAlign: "left", margin: "20px" }}>
                            Matches</h1>
                    </div>
                </div>
                <div>
                    <div className="landing-container">
                        <div className="calander">
                            <Calendar onChange={this.onChange} />
                        </div>
                        {this.props.isLoading ? (
                            <div style={{ margin: "auto" }}>
                                <Loader
                                    type="TailSpin"
                                    color="#2980b9"
                                    height="100"
                                    width="100"
                                />
                            </div>
                        ) : (
                                < div >
                                    {
                                        this.props.matches.length === 0 ? (
                                            <h2 className="h1-match" style={{ textAlign: "left", margin: "20px" }}>
                                                No Matches </h2>
                                        ) : (<div className="all-matches-box">
                                            {this.props.matches.map(match => (
                                                <div className="each-matches-box" onClick={() => {
                                                    this.props.history.push(
                                                        "/match/details/" + match.match_id,
                                                        {
                                                            match
                                                        }
                                                    );
                                                }}>
                                                    <span style={{ fontSize: "90%", margin: "1% 0 0 0" }}>{match.match_type}</span>
                                                    <div className="Team-data" style={{ fontWeight: "400" }}>
                                                        <div style={{ margin: "3% 0 0 3%" }}>{match.teamOne}</div>
                                                        <div style={{ margin: "3% 0 0 10%" }}>
                                                            {match.teamOneScore}/{match.teamone_wicket} ({match.match_values})
                                         </div>
                                                        <div style={{ margin: "3% 0 0 3%" }}>{match.teamTwo}</div>
                                                        <div style={{ margin: "3% 0 0 10%" }}>
                                                            {match.teamTwoScore}/{match.teamtwo_wicket} ({match.match_values})
                                        </div>
                                                    </div>
                                                    <span style={{ fontSize: "100%", fontWeight: "500", margin: "2% 0 0 0" }} >
                                                        {match.team_winner} won
                                    </span>
                                                </div>
                                            ))}
                                        </div>)
                                    }
                                </div>)}

                    </div>
                </div>


            </div >
        );
    }
}

const mapStateToProps = state => ({
    matches: state.matchreducer.matches,
    date: state.matchreducer.match_date,
    isLoading: state.LoadingReducer.isLoading
});

export default connect(
    mapStateToProps,
    { getRecentMatches, getMatchesDate }
)(MatchLandingPage);
