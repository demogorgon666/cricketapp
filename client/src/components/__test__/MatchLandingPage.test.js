import React from "react";
import { shallow } from "enzyme";
import { MatchLandingPage } from "../MatchLandingPage";


const getRecentMatches = jest.fn();
const getMatchesDate = jest.fn();
let matches = [];
const wrapper = shallow(

    <MatchLandingPage matches={matches} getRecentMatches={getRecentMatches} getMatchesDate={getMatchesDate} />
);

describe("Testing of MatchLandingPage Component", () => {

    it("should render the component", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("should have one h1 header", () => {
        expect(wrapper.find("h1").length).toBe(1);
    });

    it("should have one h1 content with 'Matches' content", () => {
        expect(wrapper.find(".h1-match").text()).toBe("Matches");
    });

    it("should have one h2 header", () => {
        expect(wrapper.find("h2").length).toBe(1);
    });

    it("should have 1 div tag with classname as 'timeline", () => {
        expect(wrapper.find('.timeline').length).toBe(1);
    });

    it(" [Scroller -> timeline] should have exactly defined classname", () => {
        expect(
            wrapper
                .find("div")
                .at(4)
                .props().className
        ).toBe("timeline");
    });

    it(" [inside-recent-matches-box] should have exactly defined classname", () => {
        expect(
            wrapper
                .find("div")
                .at(6)
                .props().className
        ).toBe("inside-recent-matches-box");
    });

    it(" [inside-recent-matches-box] should have exactly defined classname", () => {
        expect(
            wrapper
                .find("span")
                .at(7)
                .props().className
        ).toBe("tournamnet-name");
    });

    it(" [Team-data] should have exactly defined classname", () => {
        expect(
            wrapper
                .find("span")
                .at(8)
                .props().className
        ).toBe("Team-data");
    });
    it(" [TeamOne-name] should have exactly defined classname", () => {
        expect(
            wrapper
                .find("div")
                .at(9)
                .props().className
        ).toBe("TeamOne-name");
    });
    it(" [TeamOne-score] should have exactly defined classname", () => {
        expect(
            wrapper
                .find("div")
                .at(10)
                .props().className
        ).toBe("TeamOne-score");
    });

});
