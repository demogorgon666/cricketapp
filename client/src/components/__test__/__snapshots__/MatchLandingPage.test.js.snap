// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of MatchLandingPage Component should render the component 1`] = `
<div>
  <div>
    <Connect(Navbar)
      changeGender={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "marginTop": "80px",
      }
    }
  >
    <div>
      <h1
        className="h1-match"
        id="title-of-match"
        style={
          Object {
            "margin": "20px",
            "textAlign": "left",
          }
        }
      >
        Matches
      </h1>
    </div>
  </div>
  <div>
    <div
      className="landing-container"
    >
      <div
        className="calendar-container"
      >
        <div
          className="calender-datepicker"
        >
          <DatePicker
            clear={true}
            includeTime={false}
            locale="en"
            onChange={[Function]}
            portal={false}
            showDefaultIcon={true}
            showMonthCnt={1}
          />
        </div>
        <div
          className="calendar-fixed"
        >
          <Calendar
<<<<<<< HEAD
            base={"2019-10-19T05:20:34.698Z"}
=======
            base={"2019-10-20T17:14:43.971Z"}
>>>>>>> 0ab3b062ba51a609960cb703269d58cf69b70d34
            onChange={[Function]}
            showMonthCnt={1}
            showToday={false}
          />
        </div>
      </div>
      <div>
<<<<<<< HEAD
        <h2
          className="h1-match"
          style={
            Object {
              "margin": "20px",
              "textAlign": "left",
            }
          }
        >
          No Matches
           
        </h2>
=======
        <div
          className="all-matches-box"
        >
          <div
            className="each-matches-box"
            id="pushing-match"
            onClick={[Function]}
          >
            <span
              id="match-type"
              style={
                Object {
                  "fontSize": "15px",
                  "margin": "5px 0 0 0",
                }
              }
            >
              ODI
            </span>
            <div
              className="Team-data"
              style={
                Object {
                  "fontWeight": "400",
                }
              }
            >
              <div
                id="team-one-name"
                style={
                  Object {
                    "margin": "10px 0 0 5px",
                  }
                }
              >
                India
              </div>
              <div
                id="team-one-score"
                style={
                  Object {
                    "margin": "10px 0 0 10px",
                  }
                }
              >
                120
                /
                5
                 (
                50 overs
                )
              </div>
              <div
                id="team-two-name"
                style={
                  Object {
                    "margin": "10px 0 0 5px",
                  }
                }
              >
                Pakistan
              </div>
              <div
                id="team-two-score"
                style={
                  Object {
                    "margin": "10px 0 0 10px",
                  }
                }
              >
                150
                /
                10
                 (
                50 overs
                )
              </div>
            </div>
            <span
              id="match-winner"
              style={
                Object {
                  "fontSize": "18px",
                  "fontWeight": "500",
                  "margin": "10px 0 0 0",
                }
              }
            >
              India
               won
            </span>
          </div>
        </div>
>>>>>>> 0ab3b062ba51a609960cb703269d58cf69b70d34
      </div>
    </div>
  </div>
</div>
`;
